docs/spec/sources.md

```diff
@@ -0,0 +1,148 @@
+# Sources
+
+A **source** is any resource that generates or imports an event and relays that
+event to another resource on the cluster via CloudEvents. Sourcing events is
+critical to developing a distributed system that reacts to events.
+
+A Source:
+
+- Represents an off or on-cluster system, service or application that produces
+  events to be consumed by a sink.
+- Produces or imports CloudEvents.
+- Sends CloudEvents to the configured **sink**.
+
+In practice, sources are an abstract concept that allow us to create declarative
+configurations through the usage of Custom Resource Definitions (CRDs) extending
+Kubernetes. Those CRDs are instantiated by creating an instance of the resource.
+It is up to the implementation of the source author to understand the best way
+to realize the source application. This could be as 1:1 deployments inside of
+Kubernetes per resource, as a single multi-tenant application, or even an
+off-cluster implementation; or all combinations in-between.
+
+There are some guidelines on implementing sources to allow cluster operators and
+tools to dynamically discover and understand source installations.
+
+## Source CRDs
+
+CRDs that are to be understood as a `source` MUST be labeled:
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  labels:
+    duck.knative.dev/source: "true" # <-- required to be a source.
+```
+
+CRDs SHOULD be added to the `sources` category:
+
+```yaml
+spec:
+  names:
+    categories:
+      - sources
+```
+
```

Comment:
We should also talk about the validation aspects and encourage proper Description for the fields and so forth also?

Reply:
added.